api/PclusterApiHandler.py

```diff
@@ -551,9 +551,16 @@ def get_instance_types():
     return {"instance_types": sorted(instance_types, key=lambda x: x["InstanceType"])}
 
 
-def _get_user_roles(decoded):
-    return decoded[USER_ROLES_CLAIM] if USER_ROLES_CLAIM in decoded else ["user"]
+def _populate_identity(identity, decoded, claims):
+    if USER_ROLES_CLAIM in decoded:
+        identity["user_roles"] = decoded[USER_ROLES_CLAIM]
+    if "username" in decoded:
+        identity["username"] = decoded["username"]
 
+    for key, value in decoded.items():
+        if key in claims:
+            identity["attributes"][key] = value
+    
```

Comment:
suggestion: since you have a list of claims to retrieve from the access/id token, here you could avoid cycling on all the elements in the JWT payload, and instead cycle over the claims list itself
```python3
for claim in claims:
  if claim in decoded:
    identity["attributes"][claim] = decoded[claim]
```